### PR TITLE
adjust upgrade doc

### PR DESCRIPTION
**Pull request by wrongerror**
Signed-off-by: wrongerror <wangyifei@kubesphere.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,13 +83,30 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 ### From OpenFunction v0.6.0 to OpenFunction v0.7.x
 
 There is an incompatible breaking change needing manual actions.
```
Comment (benjaminhuo) [inline README.md:85]:
```suggestion
There is a breaking change when upgrading from v0.6.0 to 0.7.x which requires additional manual operations.
```

File: README.md [hunk 2]
```diff
@@ -83,13 +83,30 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
...
-#### OpenFunction CRDs
+#### Uninstall the Chart
+
+To uninstall/delete the `openfunction` release:
```
Comment (benjaminhuo) [inline README.md:88]:
```suggestion
First, you'll need to uninstall the old `openfunction` release:
```

File: README.md [hunk 3]
```diff
@@ -83,13 +83,30 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
...
+```shell
+kubectl get ns -o=jsonpath='{range .items[?(@.metadata.annotations.meta\.helm\.sh/release-name=="openfunction")]}{.metadata.name}: {.status.phase}{"\n"}{end}'
+```
+
+> If the knative-serving namespace is in the terminating state for a long time, try running the following command and remove finalizers:
+```shell
+kubectl edit ingresses.networking.internal.knative.dev -n knative-serving
+```
+
+#### Upgrade OpenFunction CRDs
```
Comment (benjaminhuo) [inline README.md:103]:
```suggestion
#### Upgrade OpenFunction CRDs
Then you'll need to upgrade the new OpenFunction CRDs
```

File: README.md [hunk 4]
```diff
@@ -83,13 +83,30 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
...
 
 ```shell
 kubectl apply -f https://openfunction.sh1a.qingstor.com/crds/v0.7.0/openfunction.yaml
 ```
 
-#### Components CRDs
+#### Upgrade components CRDs
```
Comment (benjaminhuo) [inline README.md:109]:
```suggestion
#### Upgrade dependent components CRDs
You also need to upgrade the dependent components' CRDs
```

File: README.md [hunk 5]
```diff
@@ -104,6 +121,12 @@ kubectl apply -f https://openfunction.sh1a.qingstor.com/crds/v0.7.0/openfunction
     kubectl apply -f https://openfunction.sh1a.qingstor.com/crds/v0.7.0/tekton-pipelines.yaml
     ```
 
+#### Install new chart
```
Comment (benjaminhuo) [inline README.md:124]: Simply installing the new chart will not upgrade all the CRDs?

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
We can add these docs to openfunction.dev , it's a place of single source of truth

File: README.md (hunk 5 above)
Comment (wrongerror) [reply README.md:126]: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations

https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md#upgrading-chart

CRDs can be managed in templates , keda and contour both manage CRDs this way, but then `helm uninstall` will delete all custom resources(for us is Function), dapr does not manage CRDs this way. Also Knative's webhook blocks the patch operation performed by helm upgrade ，so currently CRDs are not managed in this way.

**Pull request merged**